gradio/routes.py

```diff
@@ -254,6 +259,16 @@ def file(path: str):
         def file_deprecated(path: str):
             return file(path)
 
+        @app.post("/reset/")
+        @app.post("/reset")
+        async def reset_iterator(body: ResetBody):
+            if body.session_hash not in app.iterators:
+                return {"success": False}
+            async with asyncio.Lock():
```

Comment:
I don't think `async with asyncio.Lock():` will have any effect here since each request is instantiating a new `asyncio.Lock()` object. A single `asyncio.Lock()` object should be instantiated outside of the this function and that object should be acquired here. https://docs.python.org/3/library/asyncio-sync.html